module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,63 @@
+#include "compression.hpp"
+#include <iostream>
+
+void printMap(const Bitmap& bitmap)
+{
+    for (const auto line : bitmap)
+    {
+        for (const auto pixel : line)
+        {
+            std::cout << pixel;
+        }
+        std::cout << "\n";
+    }
+}
+
+CompressedBitmap compressLine(const std::array<uint8_t, width>& line)
+{
+    CompressedBitmap result;
+    std::pair<uint8_t, uint8_t> buffer{line[0], 0};
+
+    for (const auto pixel : line)
+    {
+        if (buffer.first == pixel)
+        {
+            buffer.second++;
+        }
+        else
+        {
+            result.push_back(buffer);
+            buffer.first = pixel;
+            buffer.second = 1;
+        }
+    }
+    result.push_back(buffer);
```

Comment:
You could reconfigure this loop to call push_back only once

Reply:
You got me here .. how ?
the only idea is   do .. while loop ?